core/src/main/scala/io/finch/DecodeEntity.scala

```diff
@@ -19,25 +20,25 @@ object DecodeEntity extends HighPriorityDecode {
    */
   @inline def apply[A](implicit d: DecodeEntity[A]): DecodeEntity[A] = d
 
-  implicit val decodeString: DecodeEntity[String] = instance(s => Return(s))
+  implicit val decodeString: DecodeEntity[String] = instance(s => Right(s))
 
 }
 
 trait HighPriorityDecode extends LowPriorityDecode {
 
-  implicit val decodeInt: DecodeEntity[Int] = instance(s => Try(s.toInt))
+  implicit val decodeInt: DecodeEntity[Int] = instance(s => Try(s.toInt).toEither)
```

Comment:
We would need to use `Either` directly to avoid that extra allocations for `Try`.

Reply:
Yeah. I used `scala.util.Try` to catch exceptions. If not I might have to use plain old try/catch

```scala
try {
   Right(s.toInt)
} catch {
 case e: Throwable => Left(e)
}
```

Reply:
Yeah, that's perfect.